apps/client/src/app/components/paywall/StiggPaywall.tsx

```diff
@@ -29,70 +24,37 @@ export function Paywall({
   textOverrides?: DeepPartial<PaywallLocalization>;
 }) {
   const { stigg } = useStiggContext();
-
-  async function performCheckout(
-    plan: Plan,
-    customer: Customer,
-    currentBillingPeriod: BillingPeriod,
-    unitQuantity?: number
-  ) {
-    const checkoutResult = await checkout({
-      customerId: customer.id,
-      planId: plan.id,
-      successUrl: window.location.href,
-      cancelUrl: window.location.href,
-      billingPeriod: currentBillingPeriod,
-      unitQuantity,
-    });
-
-    if (checkoutResult) {
-      window.location.href = checkoutResult.checkoutUrl;
-    }
-
-    return checkoutResult;
-  }
-
   /**
+   *
    * We support here 2 different scenarios, when there is a billing integration we use the checkout flow,
-   * and as a fallback we call to createSubscription
+   * and as a creation of a subscription without redirection to stripe
    */
   const onSubscribe: OnPlanSelectedCallbackFn = async (args) => {
     if (onPlanSelected) {
       await onPlanSelected(args);
     } else {
-      const { customer, plan, selectedBillingPeriod, intentionType } = args;
+      const { customer, plan, selectedBillingPeriod } = args;
       const collaboratorsEntitlement = stigg.getMeteredEntitlement({
         featureId: 'feature-collaborators',
       });
       const collaboratorsUnitQuantity = collaboratorsEntitlement.currentUsage;
 
       if (customer?.id) {
-        const shouldShowCheckout =
-          !customer.hasPaymentMethod &&
-          plan.pricingType !== PricingType.Free &&
-          intentionType !== SubscribeIntentionType.START_TRIAL;
-
-        let checkout = null;
-        if (shouldShowCheckout) {
-          checkout = await performCheckout(
-            plan,
-            customer,
-            selectedBillingPeriod,
-            collaboratorsUnitQuantity
-          );
-        }
+        const checkoutResult = await provisionSubscription({
```

Comment:
That's amazing how it reduce the code here! 💪

Reply:
:+1: !!